### PR TITLE
doc: improve docs and add python document

### PR DESCRIPTION
**Pull request by tamirdavid1**

### CODE REVIEW

File: docs/instrumentations/python/enrichment.mdx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+Attribute names are lowercased strings in the form `my_application.my_attribute`, example: `my_service.user_id`.
+Read more [here](https://opentelemetry.io/docs/specs/semconv/general/attribute-naming/)
+
+To record attributes, use the `set_attribute` method on the span object.
+
+``` python
+def my_function(arg: str):
+    with tracer.start_as_current_span("my_function") as span:
+        span.set_attribute("argument_value", arg)
```
Comment (blumamir) [inline docs/instrumentations/python/enrichment.mdx:68]: `"argument_value"` might be a bit unclear IMO.
What about? `"attribute_name"`?

File: docs/instrumentations/python/enrichment.mdx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+tracer = trace.get_tracer(__name__)
+
+def my_function():
+    with tracer.start_as_current_span("my_function") as span:
+        print("Hello world!")
+```
+
+Important Notes:
+
+1. **Always End a span**:
```
[the remark this answers is not in the document]
Comment (tamirdavid1) [reply docs/instrumentations/python/enrichment.mdx:36]: When the 'with' block goes out of scope, 'span' is closed for us

File: docs/instrumentations/python/python.mdx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+- [`opentelemetry-instrumentation-grpc`](https://pypi.org/project/grpcio/) versions `grpcio ~= 1.27`
+
+### Web Servers
+- [`opentelemetry-instrumentation-aiohttp-server`](https://pypi.org/project/aiohttp/) versions `aiohttp ~= 3.0`
+- [`opentelemetry-instrumentation-wsgi`](https://docs.python.org/3/library/wsgiref.html)
+
+### Cloud Services and SDKs
+- `opentelemetry-instrumentation-aws-lambda`
+- [`opentelemetry-instrumentation-boto`](https://pypi.org/project/boto/) versions `boto~=2.0`
+- [`opentelemetry-instrumentation-boto3sqs`](https://pypi.org/project/boto3/) versions `boto3 ~= 1.0`
```
[the remark this answers is not in the document]
Comment (tamirdavid1) [reply docs/instrumentations/python/python.mdx:71]: The instrumentation allows tracing for requests made by the boto3 library to the SQS service

File: docs/instrumentations/python/python.mdx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+## Traces
+
+Odigos will automatically instrument your Python services to record and collect spans for distributed tracing, by utilizing the OpenTelemetry Python official auto Instrumentation Libraries.
+
+## Instrumentation Libraries
+
+The following Python modules will be auto instrumented by Odigos:
+
+### Database Clients, ORMs, and Data Access Libraries
+- [`opentelemetry-instrumentation-aiopg`](https://pypi.org/project/aiopg/) versions `aiopg >= 0.13.0, < 2.0.0`
```
[the remark this answers is not in the document]
Comment (tamirdavid1) [reply docs/instrumentations/python/python.mdx:24]: I want it to be similar to the way it is set within setup.py/toml/requirement file which is with the minor and patch.
Agree that it is redundant but for me make more sense

File: docs/instrumentations/python/python.mdx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+- [`opentelemetry-instrumentation-botocore`](https://pypi.org/project/botocore/) versions `botocore ~= 1.0`
+
+### Data Processing and Storage
+- [`opentelemetry-instrumentation-cassandra`](https://pypi.org/project/cassandra-driver/) versions `cassandra-driver ~= 3.25, scylla-driver ~= 3.25`
+- [`opentelemetry-instrumentation-elasticsearch`](https://pypi.org/project/elasticsearch/) versions `elasticsearch >= 6.0`
+
+### Framework and Library Utilities
+- [`opentelemetry-instrumentation-falcon`](https://pypi.org/project/falcon/) versions `falcon >= 1.4.1, < 3.1.2`
+- [`opentelemetry-instrumentation-jinja2`](https://pypi.org/project/Jinja2/) versions `jinja2 >= 2.7, < 4.0`
+- [`opentelemetry-instrumentation-tornado`](https://pypi.org/project/tornado/) versions `tornado >= 5.1.1`
```
[the remark this answers is not in the document]
Comment (tamirdavid1) [reply docs/instrumentations/python/python.mdx:81]: jinja2 - templateing framework 
falcon - moved
tornado - moved

File: docs/instrumentations/python/python.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (tamirdavid1) [reply docs/instrumentations/python/python.mdx:71]: After checking botocore is using to instrument boto3

File: docs/instrumentations/python/enrichment.mdx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    Ensure that every span is ended to appear in your trace. Defer the End method of the span to guarantee that the span is always ended, even with multiple return paths in the function.
+2. **Propagate and use a valid context object**:
+    When calling tracer.Start, use a valid context object instead of context.Background(). This makes the new span a child of the active span, ensuring it appears correctly in the trace.
+3. **Pass the context object downstream**:
+    When calling downstream functions, pass the context object returned from tracer.Start() to ensure any spans created within these functions are children of the current span. This maintains the hierarchical relationship between spans and provides a clear trace of the request flow.
```
[the remark this answers is not in the document]
Comment (blumamir) [reply docs/instrumentations/python/enrichment.mdx:41]: This is marked as resolved, but the context text is still here :)

**Pull request merged**